### PR TITLE
React to aspnet/Razor#182 fix

### PR DESCRIPTION
**Pull request by dougbu**
- use `[HtmlAttributeNotBound]` to make tag helper properties `public`
- properties remain invisible in Razor

### CODE REVIEW

**Comment by NTaylorMullen** [comment]
:shipit:

**Comment by dougbu** [comment]
c162209

File: src/Microsoft.AspNet.Mvc.TagHelpers/AnchorTagHelper.cs [hunk 1]
```diff
@@ -24,9 +24,8 @@ public class AnchorTagHelper : TagHelper
         private const string RouteAttributePrefix = "asp-route-";
         private const string Href = "href";
 
-        // Protected to ensure subclasses are correctly activated. Internal for ease of use when testing.
-        [Activate]
-        protected internal IHtmlGenerator Generator { get; set; }
+        [Activate, HtmlAttributeNotBound]
```
Comment (Eilon) [inline src/Microsoft.AspNet.Mvc.TagHelpers/AnchorTagHelper.cs]: Eh, one attribute per line maybe?

Comment (Eilon) [inline src/Microsoft.AspNet.Mvc.TagHelpers/AnchorTagHelper.cs]: (And each in their own brackets.)

Comment (dougbu) [inline src/Microsoft.AspNet.Mvc.TagHelpers/AnchorTagHelper.cs]: @Eilon is that a guideline and worth an additional commit / PR?

Comment (Eilon) [inline src/Microsoft.AspNet.Mvc.TagHelpers/AnchorTagHelper.cs]: Not a guideline, but it's what we always do. Yes, please clean up. (A simple regex can fix it.)

Comment (dougbu) [inline src/Microsoft.AspNet.Mvc.TagHelpers/AnchorTagHelper.cs]: see PR #2457

**Pull request closed without merging**